api/orb/client.go

```diff
@@ -0,0 +1,142 @@
+package orb
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/CircleCI-Public/circleci-cli/api"
+	"github.com/CircleCI-Public/circleci-cli/api/graphql"
+	"github.com/CircleCI-Public/circleci-cli/settings"
+	"github.com/pkg/errors"
+)
+
+var (
+	once        sync.Once
+	client      Client
+	clientError error
+)
+
+type clientVersion string
+
+// ConfigResponse is a structure that matches the result of the GQL
+// query, so that we can use mapstructure to convert from
+// nested maps to a strongly typed struct.
+type QueryResponse struct {
+	OrbConfig struct {
+		api.ConfigResponse
+	}
+}
+
+type Client interface {
+	OrbQuery(configPath string, ownerId string) (*api.ConfigResponse, error)
+}
+
+func GetClient(config *settings.Config) (Client, error) {
+	once.Do(func() {
+		client, clientError = newClient(config)
+	})
+
+	return client, clientError
+}
+
+func newClient(config *settings.Config) (Client, error) {
+	gql := graphql.NewClient(config.HTTPClient, config.Host, config.Endpoint, config.Token, config.Debug)
+
+	clientVersion, err := detectClientVersion(gql)
+	if err != nil {
+		return nil, err
+	}
+
+	switch clientVersion {
+	case v1_string:
+		return &v1Client{gql}, nil
+	case v2_string:
+		return &v2Client{gql}, nil
+	default:
+		return nil, fmt.Errorf("Unable to recognise your server orb API")
+	}
+}
+
+func detectClientVersion(gql *graphql.Client) (clientVersion, error) {
```

Comment:
The internals of this function aren't necessarily clear. It might be worth commenting why "handling owner ID" results in being able to differentiate between v1 and v2. What do you think?